### ISSUE TITLE
na_ontap_user: cannot add a publickey for a login account

### ISSUE THREAD

**Issue by angystardust** [ansible-collections/netapp#41]
##### SUMMARY
Actually it is possible to use **na_ontap_user** module to create a password-less login using the parameter `authentication_method: publickey ` but you cannot inject a publickey for the login account. So, I think the module should be able to let a user set the following parameters:
- publickey
- index
- comment

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
`ontap/plugins/modules/na_ontap_user.py`

##### ADDITIONAL INFORMATION

The typical use-case of a password-less ssh login is to let an application running on a server to execute some netapp commands via ssh (for ex.: snap create/vol clone/etc).
Actually, the only option to inject the publickey is using the cli or using the `na_ontap_command` module giving up on the ansible idempotecy power

```yaml
- name: Inject the publickey to myapp user
  na_ontap_command:
    command:
      - security
      - login
      - publickey
      - create
      - -vserver
      - '{{ netapp_svm_name }}'
      - -username
      - '{{ myapp }}'
      - -publickey
      - '{{ mypubkey }}'
```

**Comment by angystardust** [comment]
This issue was migrated from https://github.com/ansible/ansible/issues/49401

**Comment by carchi8py** [comment]
We have DEVOPS-3587 open for this.

**Comment by lonico** [comment]
We're adding a new module, na_ontap_publickey, targeted for 21.7.0.

**Comment by lonico** [comment]
released in 21.7.0

fixed in ansible-collections/netapp.ontap@79923933476eb3dea778d6431916a24a393c3524
